From 19f5ef04dc344a82a3831f4b16f0be75c813dbf6 Mon Sep 17 00:00:00 2001
From: David Allsopp <david.allsopp@metastack.com>
Date: Thu, 25 Aug 2022 08:48:34 +0100
Subject: [PATCH 20/24] Fixup Changes

---
 Changes | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Changes b/Changes
index 763c0fa4b7..92d0837eb6 100644
--- a/Changes
+++ b/Changes
@@ -7,8 +7,8 @@ OCaml 4.14 maintenance branch
   (David Allsopp, report by William Hu, review by Xavier Leroy and
    Sébastien Hinderer)
 
-- #11???: Thwart FMA test optimization during configure
-  (William Hu, review by ???)
+- #11487: Thwart FMA test optimization during configure
+  (William Hu, review by David Allsopp and Sébastien Hinderer)
 
 ### Bug fixes:
 
@@ -21,7 +21,7 @@ OCaml 4.14 maintenance branch
   before using it to ensure that the headers can always be used in code which
   turns on -Wundef (or equivalent).
   (David Allsopp and Nicolás Ojeda Bär, review by Nicolás Ojeda Bär and
-   Sebastien Hinderer)
+   Sébastien Hinderer)
 
 - #11314, #11416: fix non-informative error message for module inclusion
   (Florian Angeletti, report by Thierry Martinez, review by Gabriel Scherer)
-- 
2.37.0.rc2

