diff -up ocaml-4.14.0/configure.3~ ocaml-4.14.0/configure
--- ocaml-4.14.0/configure.3~	2022-09-09 02:49:07.419734453 +0200
+++ ocaml-4.14.0/configure	2022-09-09 02:50:42.046990370 +0200
@@ -2795,14 +2795,14 @@ mkexe="\$(CC) \$(OC_CFLAGS) \$(CFLAGS) \
 
 # Flags for building executable files with debugging symbols
 mkexedebugflag="-g"
-common_cflags=""
-common_cppflags=""
+common_cflags="${CFLAGS}"
+common_cppflags="${CFLAGS}"
 internal_cflags=""
 internal_cppflags=""
 ocamlc_cflags=""
 ocamlc_cppflags=""
-oc_ldflags=""
-oc_dll_ldflags=""
+oc_ldflags="${LDFLAGS}"
+oc_dll_ldflags="${LDFLAGS}"
 ostype="Unix"
 SO="so"
 toolchain="cc"
@@ -12814,7 +12814,7 @@ esac ;; #(
   *) :
     case $ocaml_cv_cc_vendor in #(
   clang-*) :
-    common_cflags="-O2 -fno-strict-aliasing -fwrapv";
+    common_cflags="-O2 -fno-strict-aliasing -fwrapv ${common_cflags}";
       internal_cflags="$cc_warnings -fno-common" ;; #(
   gcc-[012]-*) :
     # Some versions known to miscompile OCaml, e,g, 2.7.2.1, some 2.96.
@@ -12839,7 +12839,7 @@ $as_echo "$as_me: WARNING: Consider usin
 -fno-builtin-memcmp";
       internal_cflags="$cc_warnings -fexcess-precision=standard" ;; #(
   gcc-*) :
-    common_cflags="-O2 -fno-strict-aliasing -fwrapv";
+    common_cflags="-O2 -fno-strict-aliasing -fwrapv ${common_cflags}";
       internal_cflags="$cc_warnings -fno-common \
 -fexcess-precision=standard" ;; #(
   msvc-*) :
