--- ocaml-4.00.0/configure.tk	2012-09-29 14:17:01.000000000 +0200
+++ ocaml-4.00.0/configure	2012-09-29 14:18:23.155049037 +0200
@@ -1433,6 +1433,8 @@ if test $has_tk = true; then
   for tk_incs in \
     "-I/usr/local/include" \
     "-I/usr/include" \
+    "-I/usr/local/include/tcl8.6 -I/usr/local/include/tk8.6" \
+    "-I/usr/include/tcl8.6 -I/usr/include/tk8.6" \
     "-I/usr/local/include/tcl8.5 -I/usr/local/include/tk8.5" \
     "-I/usr/include/tcl8.5 -I/usr/include/tk8.5" \
     "-I/usr/local/include/tcl8.4 -I/usr/local/include/tk8.4" \
@@ -1450,6 +1452,7 @@ if test $has_tk = true; then
   if test -n "$tcl_version" && test "x$tcl_version" != "xnone"; then
     echo "tcl.h and tk.h version $tcl_version found with \"$tk_defs\"."
     case $tcl_version in
+    8.6) tclmaj=8 tclmin=6 tkmaj=8 tkmin=6 ;;
     8.5) tclmaj=8 tclmin=5 tkmaj=8 tkmin=5 ;;
     8.4) tclmaj=8 tclmin=4 tkmaj=8 tkmin=4 ;;
     8.3) tclmaj=8 tclmin=3 tkmaj=8 tkmin=3 ;;
--- ocaml-4.00.0/otherlibs/labltk/support/cltkDMain.c.tk	2012-02-10 17:15:00.000000000 +0100
+++ ocaml-4.00.0/otherlibs/labltk/support/cltkDMain.c	2012-09-29 14:19:34.638421281 +0200
@@ -16,6 +16,7 @@
 
 /* $Id: cltkDMain.c 12149 2012-02-10 16:15:24Z doligez $ */
 
+#define USE_INTERP_RESULT
 #include <unistd.h>
 #include <fcntl.h>
 #include <tcl.h>
--- ocaml-4.00.0/otherlibs/labltk/support/cltkEval.c.tk	2012-02-10 17:15:00.000000000 +0100
+++ ocaml-4.00.0/otherlibs/labltk/support/cltkEval.c	2012-09-29 14:19:53.539519676 +0200
@@ -16,6 +16,7 @@
 
 /* $Id: cltkEval.c 12149 2012-02-10 16:15:24Z doligez $ */
 
+#define USE_INTERP_RESULT
 #include <stdlib.h>
 #include <string.h>
 
--- ocaml-4.00.0/otherlibs/labltk/support/cltkMain.c.tk	2012-02-10 17:15:00.000000000 +0100
+++ ocaml-4.00.0/otherlibs/labltk/support/cltkMain.c	2012-09-29 14:20:16.205637695 +0200
@@ -16,6 +16,7 @@
 
 /* $Id: cltkMain.c 12149 2012-02-10 16:15:24Z doligez $ */
 
+#define USE_INTERP_RESULT
 #include <string.h>
 #include <tcl.h>
 #include <tk.h>
--- ocaml-4.00.0/otherlibs/labltk/support/cltkVar.c.tk	2012-02-10 17:15:00.000000000 +0100
+++ ocaml-4.00.0/otherlibs/labltk/support/cltkVar.c	2012-09-29 14:20:40.660765066 +0200
@@ -17,6 +17,7 @@
 /* $Id: cltkVar.c 12800 2012-07-30 18:59:07Z doligez $ */
 
 /* Alternative to tkwait variable */
+#define USE_INTERP_RESULT
 #include <string.h>
 #include <tcl.h>
 #include <tk.h>
--- ocaml-4.00.0/otherlibs/labltk/support/cltkWait.c.tk	2011-07-27 16:16:38.000000000 +0200
+++ ocaml-4.00.0/otherlibs/labltk/support/cltkWait.c	2012-09-29 14:21:03.854885834 +0200
@@ -16,6 +16,7 @@
 
 /* $Id: cltkWait.c 11156 2011-07-27 14:17:02Z doligez $ */
 
+#define USE_INTERP_RESULT
 #include <tcl.h>
 #include <tk.h>
 #include <mlvalues.h>
--- ocaml-4.00.0/otherlibs/labltk/support/cltkMisc.c.bero	2012-09-29 14:25:59.278424082 +0200
+++ ocaml-4.00.0/otherlibs/labltk/support/cltkMisc.c	2012-09-29 14:26:07.998469544 +0200
@@ -16,6 +16,7 @@
 
 /* $Id: cltkMisc.c 12149 2012-02-10 16:15:24Z doligez $ */
 
+#define USE_INTERP_RESULT
 #include <string.h>
 #include <tcl.h>
 #include <tk.h>
