Index: ocaml-4.00.1/site-lib-src/compiler-libs/META
===================================================================
--- ocaml-4.00.1/site-lib-src/compiler-libs/META
+++ ocaml-4.00.1/site-lib-src/compiler-libs/META	2013-01-19 19:13:43.000000000 +0000
@@ -34,6 +34,5 @@
   version = "[distributed with Ocaml]"
   description = "Toplevel interactions"
   archive(byte) = "ocamltoplevel.cma"
-  archive(native) = "ocamltoplevel.cmxa"
 )
 
