--- ocaml-4.06.0/configure.0004~	2018-03-02 18:39:15.286070388 +0100
+++ ocaml-4.06.0/configure	2018-03-02 18:44:00.375083481 +0100
@@ -416,15 +416,15 @@ esac
 
 mkexe="\$(CC) \$(CFLAGS) \$(CPPFLAGS) \$(LDFLAGS)"
 mkexedebugflag="-g"
-common_cflags=""
-common_cppflags=""
+common_cflags="${CFLAGS}"
+common_cppflags="${CFLAGS}"
 internal_cflags=""
 internal_cppflags=""
 ocamlc_cflags=""
 ocamlc_cppflags=""
 ocamlopt_cflags=""
 ocamlopt_cppflags=""
-ldflags=""
+ldflags="${LDFLAGS}"
 ostype="Unix"
 exe=""
 iflexdir=""
@@ -449,7 +449,7 @@ esac
 
 case "$ccfamily" in
   clang-*)
-    common_cflags="-O2 -fno-strict-aliasing -fwrapv";
+    common_cflags="-O2 -fno-strict-aliasing -fwrapv ${common_cflags}";
     internal_cflags="$gcc_warnings";;
   gcc-[012]-*)
     # Some versions known to miscompile OCaml, e,g, 2.7.2.1, some 2.96.
@@ -467,7 +467,7 @@ case "$ccfamily" in
 -fno-builtin-memcmp";
     internal_cflags="$gcc_warnings";;
   gcc-*)
-    common_cflags="-O2 -fno-strict-aliasing -fwrapv";
+    common_cflags="-O2 -fno-strict-aliasing -fwrapv ${common_cflags}";
     internal_cflags="$gcc_warnings";;
   *)
     common_cflags="-O";;
@@ -500,7 +500,7 @@ case "$cc,$target" in
       common_cflags="-mieee $common_cflags";
     fi
     # Put code and static data in lower 4GB
-    ldflags="-Wl,-T,12000000 -Wl,-D,14000000"
+    ldflags="${ldflags} -Wl,-T,12000000 -Wl,-D,14000000"
     # Tell gcc that we can use 32-bit code addresses for threaded code
     echo "#define ARCH_CODE32" >> m.h;;
   cc,alpha*-*-osf*)
