diff -Naur ocaml-4.07.0/asmcomp/amd64/emit.mlp ocaml-4.07.0.tpg/asmcomp/amd64/emit.mlp
--- ocaml-4.07.0/asmcomp/amd64/emit.mlp	2018-07-10 14:09:26.000000000 +0000
+++ ocaml-4.07.0.tpg/asmcomp/amd64/emit.mlp	2021-10-21 17:19:54.977046330 +0000
@@ -994,7 +994,7 @@
     | S_macosx -> D.section ["__TEXT";"__literal16"] None ["16byte_literals"]
     | S_mingw64 | S_cygwin -> D.section [".rdata"] (Some "dr") []
     | S_win64 -> D.data ()
-    | _ -> D.section [".rodata.cst8"] (Some "a") ["@progbits"]
+    | _ -> D.section [".rodata.cst16"] (Some "aM") ["@progbits";"16"]
     end;
     D.align 16;
     _label (emit_symbol "caml_negf_mask");
@@ -1054,8 +1054,9 @@
     | S_macosx -> D.section ["__TEXT";"__literal8"] None ["8byte_literals"]
     | S_mingw64 | S_cygwin -> D.section [".rdata"] (Some "dr") []
     | S_win64 -> D.data ()
-    | _ -> D.section [".rodata.cst8"] (Some "a") ["@progbits"]
+    | _ -> D.section [".rodata.cst8"] (Some "aM") ["@progbits";"8"]
     end;
+    D.align 8;
     List.iter (fun (cst,lbl) -> emit_float_constant cst lbl) !float_constants
   end;
 
diff -Naur ocaml-4.07.0/asmrun/amd64.S ocaml-4.07.0.tpg/asmrun/amd64.S
--- ocaml-4.07.0/asmrun/amd64.S	2018-07-10 14:09:26.000000000 +0000
+++ ocaml-4.07.0.tpg/asmrun/amd64.S	2021-10-21 17:21:52.377440240 +0000
@@ -724,7 +724,7 @@
 #elif defined(SYS_mingw64) || defined(SYS_cygwin)
         .section .rdata,"dr"
 #else
-        .section    .rodata.cst8,"a",@progbits
+        .section    .rodata.cst16,"aM",@progbits,16
 #endif
         .globl  G(caml_negf_mask)
         .align  SIXTEEN_ALIGN
